rust-src/wallet_library/src/statement.rs

```diff
@@ -0,0 +1,274 @@
+use concordium_base::{
+    curve_arithmetic::Curve,
+    id::{id_proof_types::AtomicStatement, types::*},
+    web3id::{CredentialStatement, Request},
+};
+
+const IDENTITY_RANGE_TAGS: &[AttributeTag] = &[AttributeTag(3), AttributeTag(9), AttributeTag(10)]; // dob, idDocIssuedAt, idDocExpiresAt
+const IDENTITY_SET_TAGS: &[AttributeTag] = &[
+    AttributeTag(4),
+    AttributeTag(5),
+    AttributeTag(6),
+    AttributeTag(8),
+]; // Country of residence, Nationality, IdDocType, IdDocIssuer
+
+fn is_iso8601(date: &str) -> bool {
+    println!("{}", date);
+    chrono::NaiveDate::parse_from_str(date, "%Y%m%d").is_ok()
+}
+
+fn is_iso3166_alpha_2(code: &str) -> bool { rust_iso3166::from_alpha2(code).is_some() }
+
+fn is_iso3166_2(code: &str) -> bool { rust_iso3166::iso3166_2::from_code(code).is_some() }
+
+pub trait WalletCheck {
+    fn satisfies_wallet_restrictions(&self) -> bool;
+}
+
+impl<C: Curve, AttributeType: Attribute<C::Scalar>> WalletCheck for Request<C, AttributeType> {
+    fn satisfies_wallet_restrictions(&self) -> bool {
+        self.credential_statements
+            .iter()
+            .all(|s| s.satisfies_wallet_restrictions())
+    }
+}
+
+impl<C: Curve, AttributeType: Attribute<C::Scalar>> WalletCheck
+    for CredentialStatement<C, AttributeType>
+{
+    fn satisfies_wallet_restrictions(&self) -> bool {
+        match self {
+            CredentialStatement::Account {
+                statement,
+                network: _,
+                cred_id: _,
+            } => {
+                if statement.is_empty() {
+                    return false;
+                }
+                let mut used_tags = Vec::<AttributeTag>::new();
+                for atomic_statement in statement {
+                    if used_tags.contains(&atomic_statement.attribute()) {
+                        return false;
+                    }
+                    used_tags.push(atomic_statement.attribute());
```

Comment:
Please use a set here. Don't be [accidentally quadratic](https://accidentallyquadratic.tumblr.com/)